### PR TITLE
Move particle serialization into ParticleHandler

### PR DESCRIPTION
**Pull request by gassmoeller**
This moves most of the particle serialization functionality into the ParticleHandler class. Some parts remain in World for now, I will address them in a follow-up PR.

### CODE REVIEW

**Comment by bangerth** [comment]
There is one failing test. Is this expected?

File: source/particle/world.cc [hunk 1]
```diff
@@ -148,125 +143,66 @@ namespace aspect
...
-    void
-    World<dim>::register_load_callback_function(const bool serialization,
-                                                typename parallel::distributed::Triangulation<dim> &triangulation)
-    {
-      TimerOutput::Scope timer_section(this->get_computing_timer(), "Particles: Refine mesh, load");
+//      const parallel::distributed::Triangulation<dim> &triangulation = this->get_triangulation();
+//      parallel::distributed::Triangulation<dim> &non_const_triangulation =
+//        const_cast<parallel::distributed::Triangulation<dim> &> (triangulation);
+//
+//#if DEAL_II_VERSION_GTE(9,0,0)
```
Comment (bangerth) [inline source/particle/world.cc]: do you want to get these commented out lines back in?

File: source/particle/world.cc [hunk 2]
```diff
@@ -148,125 +143,66 @@ namespace aspect
...
-            = std_cxx11::bind(&aspect::Particle::ParticleHandler<dim>::load_particles,
-                              std_cxx11::ref(*particle_handler),
-                              std_cxx11::_1,
-                              std_cxx11::_2,
-                              std_cxx11::_3);
 
-          triangulation.notify_ready_to_unpack(data_offset,callback_function);
 
-          apply_particle_per_cell_bounds();
+//#endif
```
Comment (bangerth) [inline source/particle/world.cc]: and here?

**Comment by gassmoeller** [comment]
> There is one failing test. Is this expected?

No that was a bug, although just a missing update of a cached number. Fixed.
The commented lines sneaked in there unintentionally, I removed them. They do not work yet, and can only be used when dealii/dealii#5150 is merged anyway.

**Comment by gassmoeller** [comment]
And it is simpler to read the patch in split mode. I mostly just move functions from World to ParticleHandler (and adjust some minor things that are connected to this move).

**Pull request merged**